### PR TITLE
[ticket/12692] Add a console command to manage the thumbnails

### PR DESCRIPTION
**Pull request by Nicofuma**
https://tracker.phpbb.com/browse/PHPBB3-12692

PHPBB3-12692

### CODE REVIEW

**Comment by CHItA** [comment]
:+1:

File: phpBB/phpbb/console/command/thumbnail/delete.php [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	/**
+	* phpBB root path
+	* @var string
+	*/
+	protected $phpbb_root_path;
+
+	/**
+	* Constructor
+	*
+	* @param \phpbb\user $user The user object (used to get language information)
```
Comment (marc1706) [inline phpBB/phpbb/console/command/thumbnail/delete.php]: I guess you're using $user instead of the language class due to the parent constructor?

File: phpBB/phpbb/console/command/thumbnail/delete.php [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	/**
+	* Sets the command name and description
+	*
+	* @return null
+	*/
+	protected function configure()
+	{
+		$this
```
Comment (marc1706) [inline phpBB/phpbb/console/command/thumbnail/delete.php]: I'd like this better:
```
$this->setName('thumbnail:delete')
			->setDescription($this->user->lang('CLI_DESCRIPTION_THUMBNAIL_DELETE'));
```

File: phpBB/phpbb/console/command/thumbnail/generate.php [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	/**
+	* Sets the command name and description
+	*
+	* @return null
+	*/
+	protected function configure()
+	{
+		$this
```
Comment (marc1706) [inline phpBB/phpbb/console/command/thumbnail/generate.php]: See above comment.

File: phpBB/phpbb/console/command/thumbnail/recreate.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+class recreate extends \phpbb\console\command\command
+{
+	/**
+	* Sets the command name and description
+	*
+	* @return null
+	*/
+	protected function configure()
+	{
+		$this
```
Comment (marc1706) [inline phpBB/phpbb/console/command/thumbnail/recreate.php]: and here :)

File: phpBB/phpbb/console/command/thumbnail/delete.php (hunk 1 above)
Comment (Nicofuma) [inline phpBB/phpbb/console/command/thumbnail/delete.php]: yep

File: tests/console/thumbnail_test.php [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+		$this->application->add(new generate($this->user, $this->db, $this->cache, $this->phpbb_root_path, $this->phpEx));
+		$this->application->add(new delete($this->user, $this->db, $this->phpbb_root_path));
+		$this->application->add(new recreate($this->user));
+
+		$phpbb_filesystem = new \phpbb\filesystem\filesystem();
+	}
+
+	public function test_thumbnails()
+	{
+		copy(dirname(__FILE__) . '/fixtures/png.png', $this->phpbb_root_path . 'files/test_png_1');
```
Comment (marc1706) [inline tests/console/thumbnail_test.php]: Should probably do this in the setUp method and remove those images in the tearDown method.
Also, you should be able to use dataproviders for this instead of repeating yourself.

File: phpBB/phpbb/console/command/thumbnail/delete.php (hunk 2 above)
Comment (Nicofuma) [inline phpBB/phpbb/console/command/thumbnail/delete.php]: That's what we did for the other commands

File: phpBB/language/en/cli.php [hunk 6]
```diff
@@ -80,13 +84,23 @@
 	'CLI_EXTENSION_PURGE_FAILURE'		=> 'Could not purge extension %s',
 	'CLI_EXTENSION_PURGE_SUCCESS'		=> 'Successfully purged extension %s',
 	'CLI_EXTENSION_NOT_FOUND'			=> 'No extensions were found.',
-	'CLI_EXTENSIONS_AVAILABLE'			=> 'Available',
```
Comment (Elsensee) [inline phpBB/language/en/cli.php]: Why did you remove them? They are still in use...

File: tests/console/thumbnail_test.php (hunk 5 above)
Comment (Nicofuma) [inline tests/console/thumbnail_test.php]: I can't use dataproviders because I'm using the state of the previous test

updated

File: phpBB/language/en/cli.php (hunk 6 above)
Comment (Nicofuma) [inline phpBB/language/en/cli.php]: fixed
